scripts/dts/extract/globals.py

```diff
@@ -294,7 +294,8 @@ def get_binding(node_path):
         parent_binding = bindings[parent_compat]
         if 'child' in parent_binding and 'bus' in parent_binding['child']:
             bus = parent_binding['child']['bus']
-            return bus_bindings[bus][compat]
+            if bus in bus_bindings and compat in bus_bindings[bus]:
+                return bus_bindings[bus][compat]
```

Comment:
Why not just add the bus binding?

Reply:
Not really sure why this was needed?

Reply:
@henrikbrixandersen This was not needed until I rebased onto [this commmit](https://github.com/zephyrproject-rtos/zephyr/commit/f9dd53624c578be51fbb91ddf2938df55a0c6966).

With the new version of `globals.py`, sanitycheck fails with the following error:

```
Zephyr version: 1.14.99
-- Found PythonInterp: /usr/bin/python3 (found suitable version "3.7.3", minimum required is "3.4") 
-- Selected BOARD hifive1
-- Found west: /usr/local/bin/west (found suitable version "0.5.8", minimum required is "0.5.6")
-- Loading litex-zephyr/boards/riscv32/hifive1/hifive1.dts as base
-- Overlaying litex-zephyr/dts/common/common.dts
Traceback (most recent call last):
  File "litex-zephyr/scripts/dts/extract_dts_includes.py", line 517, in <module>
    main()
  File "litex-zephyr/scripts/dts/extract_dts_includes.py", line 503, in main
    generate_defines()
  File "litex-zephyr/scripts/dts/extract_dts_includes.py", line 446, in generate_defines
    generate_node_defines(node_path)
  File "litex-zephyr/scripts/dts/extract_dts_includes.py", line 111, in generate_node_defines
    generate_bus_defines(node_path)
  File "litex-zephyr/scripts/dts/extract_dts_includes.py", line 140, in generate_bus_defines
    binding = get_binding(node_path)
  File "litex-zephyr/scripts/dts/extract/globals.py", line 297, in get_binding
    return bus_bindings[bus][compat]
KeyError: 'soc-nv-flash'
CMake Error at litex-zephyr/cmake/dts.cmake:174 (message):
  command failed with return code: 1
Call Stack (most recent call first):
  litex-zephyr/cmake/app/boilerplate.cmake:403 (include)
  CMakeLists.txt:4 (include)


-- Configuring incomplete, errors occurred!
```

Having in mind that:

-  it worked before the rebase, 
- seeing [a comment](https://github.com/zephyrproject-rtos/zephyr/blob/master/scripts/dts/extract/globals.py#L299) and 
- seeing that all previous accesses to the dictionaries are protected, 

I assumed that the code should fall-back to [the final return](https://github.com/zephyrproject-rtos/zephyr/blob/master/scripts/dts/extract/globals.py#L300) and not throw an exception on a dictionary access.

Reply:
@MaureenHelm I'm affraid I don't follow. I'm not very fluent in the dts-extraction code, so if there is a better place where the fix should go or the DTS itself should be reworked, please provide me with more details.

Reply:
@mateusz-holenko Ahh, now I see it. It is due to you using `compatible = "soc-nv-flash"` for something, that's an external SPI flash, so the DTS YAML binding does not include the SPI device bus setup (as defined in `spi-device.yaml `).

I am not familiar with what kind of flash is present on the HiFive1. Can you use `compatible = "jedec,spi-nor"` instead?

Reply:
@henrikbrixandersen Thanks for the hint.

According to [this doc](https://sifive.cdn.prismic.io/sifive%2F7503d08c-5f6f-432e-a721-bb02f75cb5e7_hifive1-a01-bom.pdf) (position 43), this is [IS25LP128-JBLE](https://eu.mouser.com/datasheet/2/198/IS25LP032-064-128-463542.pdf) flash. [Linux driver](https://github.com/torvalds/linux/blob/master/drivers/mtd/spi-nor/spi-nor.c#L1826) uses `0x9d6019` as JEDEC ID.

My proposition is to describe it as:
```
      flash0: flash@0 {
              compatible = "issi,is25lp128", "jedec,spi-nor";
              jedec-id = <0x96 0x60 0x18>;
              reg = <0>;
      };
```

With this shape of the DTS my change in `globals.py` is not needed anymore.

Reply:
Ok, I did that. Does this work for you @MaureenHelm?